scripts/pre_commit_linter.py

```diff
@@ -2147,42 +2166,103 @@ def _check_for_copyright_notice(self):
 
         return summary_messages
 
+    def check_for_important_rules_at_bottom_of_codeowners(
+                self, imp_rules_in_critical_section, failed):
+        """Checks that the most important rules/patterns are at the bottom
+        of the CODEOWNERS file.
+
+        Arguments:
+            imp_rules_in_critical_section: list(str). List of the important
+                paths/rules for CODEOWNERS file.
+            failed: bool. Current status of failure.
+
+        Returns:
+            bool. Status of failure.
+        """
+
+        # The following condition checks for extra rules in the critical
+        # files section in the .github/CODEOWNERS file.
+        if len(imp_rules_in_critical_section) > len(
+                CODEOWNER_IMPORTANT_PATHS):
+            important_path_match_bool_list = ([
+                imp_path in CODEOWNER_IMPORTANT_PATHS
+                for imp_path in imp_rules_in_critical_section])
+            for index, bool_value in enumerate(
+                    important_path_match_bool_list):
+                if not bool_value:
+                    print ('%s --> The rule \'%s\' is not an important '
+                            'rule. Please place it before the critical '
+                            'files section.' % (
+                                CODEOWNER_FILEPATH,
+                                imp_rules_in_critical_section[index]))
+                    failed = True
+        else:
+            important_path_match_bool_list = ([
+                imp_path in imp_rules_in_critical_section
+                for imp_path in CODEOWNER_IMPORTANT_PATHS])
+            # This condition checks that all the values in the boolean list
+            # are True. This ensures that the critical files of the
+            # CODEOWNERS file matches with the CODEOWNER_IMPORTANT_PATHS.
+            for index, bool_value in enumerate(
+                    important_path_match_bool_list):
+                if not bool_value:
+                    print ('%s --> Please ensure that the rule \'%s\' lies '
```

Comment:
Do we have to print this through _redirect_stdout? (not sure!)

Reply:
No I dont think so since all the print statements are printed without _redirect_stdout.